EBNF.Compiler/Compiler.cs

```diff
@@ -203,6 +203,39 @@ public static string[] CompileEbnf(string rawData, string namespaceName)
                         indentLevel++;
                         code.Add($"{new string('\t', 3 + indentLevel)}remainder = remainder.Substring({concatenation.Length - 2});");
                         break;
+                    case '!': // Regex Value
+                        
+                        code.Add($"var regex = new Regex({concatenation.Remove(0, 1)});");
+                        code.Add($"{new string('\t', 3 + indentLevel)}var match = regex.Match(remainder);");
+                        code.Add($"{new string('\t', 3 + indentLevel)}if (match.Success)");
+                        code.Add($"{new string('\t', 3 + indentLevel)}{{");
+                        indentLevel++;
+                        code.Add($"{new string('\t', 3 + indentLevel)}if (remainder.StartsWith(\" \"))");
+                        code.Add($"{new string('\t', 3 + indentLevel)}" + "{");
+                        indentLevel++;
+                        code.Add($"{new string('\t', 3 + indentLevel)}remainder = remainder.Remove(0, match.Groups[0].Value.Length + 2);");
```

Comment:
If I understand your code correctly here, I believe you shouldn't be only considering the first group of a single match.

Please read the examples given [here](https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.match.groups?view=netframework-4.7.2#System_Text_RegularExpressions_Match_Groups) as I believe you have misunderstood the meaning of a group as opposed to a match.

Also, why the +2 here? Perhaps the requirement for this will be resolved when Groups[0] is removed.

Finally, please use String.Substring for the reasons stated above.

```suggestion
                        code.Add($"{new string('\t', 3 + indentLevel)}remainder = remainder.Substring(match.Length + 2);");
```